crates/core/c8y_smartrest/src/operations.rs

```diff
@@ -60,13 +61,40 @@ impl Default for Operations {
 
 impl Operations {
     pub fn add(&mut self, operation: Operation) {
-        if let Some(detail) = operation.exec() {
-            if let Some(on_message) = &detail.on_message {
-                self.operations_by_trigger
-                    .insert(on_message.clone(), self.operations.len());
+        if self.operations.iter().any(|o| o.name.eq(&operation.name)) {
+            return;
+        } else {
+            if let Some(detail) = operation.exec() {
+                if let Some(on_message) = &detail.on_message {
+                    self.operations_by_trigger
+                        .insert(on_message.clone(), self.operations.len());
+                }
             }
+            self.operations.push(operation);
+        }
+    }
+
+    pub fn add_operation(&mut self, operation: Operation) -> Result<(), OperationsError> {
+        match operation.topic() {
+            Some(topic) => {
+                if topic.eq(&String::from("c8y/s/ds")) {
+                    self.add(operation);
+                } else {
+                    warn!("The operation {} is not supported", operation.name);
+                }
```

Comment:
So, you mean to add the operations without checking the topic of the operation?.

> Please consider to
> 
> * use a `tokio::select` in the main mapper loop to select the first event coming either from inotify and mqtt,
> * either react on `CLOSE_WRITE` or have a reaction that doesn't depend on the content of the file.
Reacting on `CLOSE_WRITE` make sense. We can not avoid content parsing because the content contains the `plugin path`.